core/src/test/kotlin/org/stellar/anchor/sep6/ExchangeAmountsCalculatorTest.kt

```diff
@@ -0,0 +1,150 @@
+package org.stellar.anchor.sep6
+
+import io.mockk.MockKAnnotations
+import io.mockk.clearAllMocks
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.unmockkAll
+import kotlin.test.assertEquals
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.stellar.anchor.TestConstants.Companion.TEST_ACCOUNT
+import org.stellar.anchor.TestConstants.Companion.TEST_ASSET
+import org.stellar.anchor.TestConstants.Companion.TEST_ASSET_SEP38_FORMAT
+import org.stellar.anchor.api.callback.FeeIntegration
+import org.stellar.anchor.api.callback.GetFeeRequest
+import org.stellar.anchor.api.callback.GetFeeResponse
+import org.stellar.anchor.api.exception.BadRequestException
+import org.stellar.anchor.api.exception.SepValidationException
+import org.stellar.anchor.api.sep.sep38.RateFee
+import org.stellar.anchor.api.shared.Amount
+import org.stellar.anchor.asset.AssetService
+import org.stellar.anchor.asset.DefaultAssetService
+import org.stellar.anchor.sep38.PojoSep38Quote
+import org.stellar.anchor.sep38.Sep38QuoteStore
+import org.stellar.anchor.sep6.ExchangeAmountsCalculator.Amounts
+
+class ExchangeAmountsCalculatorTest {
+  private val assetService: AssetService = DefaultAssetService.fromJsonResource("test_assets.json")
+
+  @MockK(relaxed = true) lateinit var feeIntegration: FeeIntegration
+  @MockK(relaxed = true) lateinit var sep38QuoteStore: Sep38QuoteStore
+
+  private lateinit var calculator: ExchangeAmountsCalculator
+
+  @BeforeEach
+  fun setup() {
+    MockKAnnotations.init(this, relaxUnitFun = true)
+    calculator = ExchangeAmountsCalculator(feeIntegration, sep38QuoteStore, assetService)
+  }
+
+  @AfterEach
+  fun teardown() {
```

Comment:
This can hurt for parallel testing.

It is safer to only unmock necessary objects.